macOS: Allow setting some fullscreen style options

TcT2k opened the pull request:
When using the native fullscreen API by enabling `EnableFullScrenView()`
allow using hiding (or showing) menu and/or toolbar. An additional style
parameter has been added to `EnableFullScrenView()` to allow customizing
which style is applied when the user presses the fullscreen button
instead of a call to `ShowFullScreen()`.

As an example the webview sample previously would still have the toolbar visible in fullscreen. It now auto hides which is the common behaviour for macOS apps (yet it isn't the system default API behavior).

vadz commented on interface/wx/toplevel.h:
```diff
@@ -649,6 +649,10 @@ class wxTopLevelWindow : public wxNonOwnedWindow
+        @param style
+            This parameter sets which elements will be hidden when the
+            user presses the full screen button. See ShowFullScreen()
+            for possible values.
```
```suggestion
            for possible values. It is available since wxWidgets 3.1.6.
```

We probably should also mention in `wxFULLSCREEN_XXX` documentation which ones of them apply to `EnableFullScreenView()` too, and which ones are for `ShowFullScreen()` only.

vadz reviewed (commented):
Thanks, no real comments and I'll merge this soon if @csomor doesn't have any objections.

TcT2k commented:
> Thanks, no real comments and I'll merge this soon if @csomor doesn't have any objections.

Merged your changes and documented which flags can be used with the native full screen API.

TcT2k committed:
macOS: Allow setting some fullscreen style options

When using the native fullscreen API by enabling EnableFullScrenView()
allow using hiding (or showing) menu and/or toolbar. An additional style
parameter has been added to EnableFullScrenView() to allow customizing
which style is applied when the user presses the fullscreen button
instead of a call to ShowFullScreen().

vadz commented on src/osx/cocoa/nonownedwnd.mm:
```diff
@@ -335,6 +337,27 @@ - (id) init
 }
 
+- (NSApplicationPresentationOptions)window:(NSWindow *)window
+      willUseFullScreenPresentationOptions:(NSApplicationPresentationOptions)proposedOptions
```
Looking at [the docs](https://developer.apple.com/documentation/appkit/nswindowdelegate/1419144-window?language=objc) it seems like we are supposed to use `proposedOptions` by default, so I wonder if it's correct to just ignore them?

No other comments from me and if you think this is fine and if Stefan doesn't say anything neither, I'll merge this soon.

Thanks again!

TcT2k replied:
I think it's OK not to use the `proposedOptions` as we don't implement presentation options in the application delegate (which would be the source for the `proposedOptions` value if I'm not mistaken).
